with-astro/src/pages/index.astro

```diff
@@ -0,0 +1,36 @@
+---
+import Layout from '../layouts/default.astro'
+
+const response = await fetch(
+  'https://api-eu-central-1.graphcms.com/v2/ck8sn5tnf01gc01z89dbc7s0o/master',
+  {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+    body: JSON.stringify({
+      query: `
+        {
+          products {
+            slug
+            name
+          }
+        }
+      `,
+    }),
+  }
+)
```

Comment:
Maybe we could wrap this fetch in a separate file, and import it here and in the other files that has the `fetch`.

Reply:
Think this will be fine as is, simple example to demonstrate how to use it